tests/pytests/unit/states/test_esxdatacenter.py

```diff
@@ -0,0 +1,237 @@
+"""
+    :codeauthor: :email:`Alexandru Bleotu <alexandru.bleotu@morganstanley.com>`
+"""
+import pytest
+import salt.states.esxdatacenter as esxdatacenter
+from salt.exceptions import CommandExecutionError
+from tests.support.mock import MagicMock, patch
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {esxdatacenter: {}}
+
+
+@pytest.fixture(scope="function")
+def policy_set():
+    try:
+        patcher = patch.dict(esxdatacenter.__opts__, {"test": False})
+        patcher.start()
+
+        mock_si = MagicMock()
+        mock_dc = MagicMock()
+
+        with patch.dict(
+            esxdatacenter.__salt__,
+            {
+                "vsphere.get_proxy_type": MagicMock(),
+                "vsphere.get_service_instance_via_proxy": MagicMock(
+                    return_value=mock_si
+                ),
+                "vsphere.list_datacenters_via_proxy": MagicMock(return_value=[mock_dc]),
+                "vsphere.disconnect": MagicMock(),
+            },
+        ):
+            yield
+    finally:
+        mock_si = MagicMock()
+        mock_dc = MagicMock()
+
+        patch.dict(
+            esxdatacenter.__salt__,
+            {
+                "vsphere.get_proxy_type": MagicMock(),
+                "vsphere.get_service_instance_via_proxy": MagicMock(
+                    return_value=mock_si
+                ),
+                "vsphere.list_datacenters_via_proxy": MagicMock(return_value=[mock_dc]),
+                "vsphere.disconnect": MagicMock(),
+            },
+        )
```

Comment:
```suggestion
@pytest.fixture(scope="function")
def policy_set():
    patcher_1 = patcher_2 = None
    try:
        patcher_1 = patch.dict(esxdatacenter.__opts__, {"test": False})
        patcher_1.start()

        mock_si = MagicMock()
        mock_dc = MagicMock()

        patcher_2 = patch.dict(
            esxdatacenter.__salt__,
            {
                "vsphere.get_proxy_type": MagicMock(),
                "vsphere.get_service_instance_via_proxy": MagicMock(
                    return_value=mock_si
                ),
                "vsphere.list_datacenters_via_proxy": MagicMock(return_value=[mock_dc]),
                "vsphere.disconnect": MagicMock(),
            },
        )
        patcher_2.start()
        yield
    finally:
        if patcher_1:
            patcher_1.stop()
        if patcher_2:
            patcher_2.stop()
```
This way it should still work, can you please try?

Reply:
Yep, it works, applied these changes